### PR TITLE
treewide: move pkg-config into nativeBuildInputs

### PR DESCRIPTION
**Pull request by siraben**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by roberth** [comment]
This doesn't make much sense to me if you don't enable `strictDeps` as well.

File: pkgs/applications/networking/sync/lsyncd/default.nix [hunk 1]
```diff
@@ -31,9 +31,10 @@ stdenv.mkDerivation rec {
 
   dontUseCmakeBuildDir = true;
 
+  nativeBuildInputs = [ pkg-config ];
   buildInputs = [
     rsync
-    cmake lua pkg-config
+    cmake lua
     asciidoc libxml2 docbook_xml_dtd_45 docbook_xsl libxslt
```
Comment (jtojnar) [inline pkgs/applications/networking/sync/lsyncd/default.nix:38]: Almost everything but lua and rsync should probably go to native.

File: pkgs/applications/science/biology/vcftools/default.nix [hunk 2]
```diff
@@ -11,7 +11,8 @@ stdenv.mkDerivation rec {
     sha256 = "0msb09d2cnm8rlpg8bsc1lhjddvp3kf3i9dsj1qs4qgsdlzhxkyx";
   };
 
-  buildInputs = [ autoreconfHook pkg-config zlib perl ];
+  nativeBuildInputs = [ pkg-config ];
```
Comment (jtojnar) [inline pkgs/applications/science/biology/vcftools/default.nix:14]: `autoreconfHook` too.

File: pkgs/applications/terminal-emulators/mrxvt/default.nix [hunk 3]
```diff
@@ -22,7 +22,8 @@ stdenv.mkDerivation rec {
     sha256 = "1mqhmnlz32lvld9rc6c1hyz7gjw4anwf39yhbsjkikcgj1das0zl";
   };
 
-  buildInputs = [ libX11 libXft libXi xorgproto libSM libICE freetype pkg-config which ];
+  nativeBuildInputs = [ pkg-config ];
```
Comment (jtojnar) [inline pkgs/applications/terminal-emulators/mrxvt/default.nix:25]: Probably also `which`

File: pkgs/development/compilers/aliceml/default.nix [hunk 4]
```diff
@@ -16,12 +16,13 @@ stdenv.mkDerivation {
     sha256 = "0mgc6llbq166jmlq3alvagqsg3730670zvbwwkdgsqklw70v9355";
   };
 
+  nativeBuildInputs = [ pkg-config ];
   buildInputs = [
     stdenv gcc glibc
     libtool gnumake autoconf automake
     file which zsh m4 gtk2 zlib gmp
```
Comment (jtojnar) [inline pkgs/development/compilers/aliceml/default.nix:23]: libtool to which probably too.

File: pkgs/development/compilers/intercal/default.nix [hunk 5]
```diff
@@ -14,8 +14,9 @@ stdenv.mkDerivation rec {
     sha256 = "1z2gpa5rbqb7jscqlf258k0b0jc7d2zkyipb5csjpj6d3sw45n4k";
   };
 
+  nativeBuildInputs = [ pkg-config ];
   buildInputs =
-  [ pkg-config bison flex makeWrapper ];
+  [ bison flex makeWrapper ];
```
Comment (jtojnar) [inline pkgs/development/compilers/intercal/default.nix:19]: All of these.

File: pkgs/development/libraries/template-glib/default.nix [hunk 6]
```diff
@@ -13,8 +13,8 @@ stdenv.mkDerivation {
     sha256 = "nsm3HgTU9csU91XveQYxzQtFwGA+Ecg2/Hz9niaM0Ho=";
   };
 
-  buildInputs = [ meson ninja pkg-config gettext flex bison vala glib gtk-doc docbook_xsl docbook_xml_dtd_43 ];
-  nativeBuildInputs = [ glib gobject-introspection ];
+  buildInputs = [ meson ninja gettext flex bison vala glib gtk-doc docbook_xsl docbook_xml_dtd_43 ];
```
Comment (jtojnar) [inline pkgs/development/libraries/template-glib/default.nix:16]: See https://github.com/NixOS/nixpkgs/pull/166511#discussion_r839343355

File: pkgs/development/libraries/xcb-util-cursor/HEAD.nix [hunk 7]
```diff
@@ -21,14 +21,14 @@ stdenv.mkDerivation {
 
   outputs = [ "out" "dev" ];
 
+  nativeBuildInputs = [ pkg-config ];
   buildInputs = [
     autoconf
     automake
     gnum4
     gperf
     libtool
```
Comment (jtojnar) [inline pkgs/development/libraries/xcb-util-cursor/HEAD.nix:30]: These too.

File: pkgs/development/tools/misc/distcc/default.nix [hunk 8]
```diff
@@ -18,7 +18,7 @@ let
     };
 
   nativeBuildInputs = [ pkg-config ];
-    buildInputs = [popt avahi pkg-config python3 gtk3 autoconf automake which procps libiberty_static];
+    buildInputs = [popt avahi python3 gtk3 autoconf automake which procps libiberty_static];
```
Comment (jtojnar) [inline pkgs/development/tools/misc/distcc/default.nix:21]: `autoconf automake which`

File: pkgs/games/extremetuxracer/default.nix [hunk 9]
```diff
@@ -13,11 +13,12 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-ktqWPtMqub/xJjRu37lUnocdfDzsdT9KxQmODVPDg0E=";
   };
 
+  nativeBuildInputs = [ pkg-config ];
   buildInputs = [
     libGLU libGL libX11 xorgproto tcl freeglut freetype
     sfml libXi
     libXmu libXext libXt libSM libICE
-    libpng pkg-config gettext intltool
+    libpng gettext intltool
```
Comment (jtojnar) [inline pkgs/games/extremetuxracer/default.nix:21]: `gettext intltool`

File: pkgs/games/freedink/default.nix [hunk 10]
```diff
@@ -25,9 +25,10 @@ in stdenv.mkDerivation rec {
     sha256 = "19xximbcm6506kvpf3s0q96697kmzca3yrjdr6dgphklp33zqsqr";
   };
 
+  nativeBuildInputs = [ pkg-config ];
   buildInputs = [
     SDL SDL_mixer SDL_image SDL_ttf SDL_gfx
-    pkg-config intltool fontconfig libzip zip zlib
+    intltool fontconfig libzip zip zlib
```
Comment (jtojnar) [inline pkgs/games/freedink/default.nix:31]: `intltool`

**Comment by siraben** [comment]
@jtojnar usually with these treewide PRs I target one specific package that should go into `nativeBuildInputs`, do you think I should just do a list of all of them?

**Comment by siraben** [comment]
@roberth turning `strictDeps` on risks further breakage on native -> native compilation, and I can't necessarily test every package I'm changing here because I don't have a beefy `x86_64-linux` machine ATM.

**Comment by jtojnar** [comment]
Yeah, I think a list is better for avoiding diff noise.

**Comment by roberth** [comment]
Doesn't seem like time well spent if the packages can regress, but you do you. These changes are still improvements. I'll leave you to it.

**Comment by siraben** [comment]
I will make a PR that makes multiple changes into nativeBuildInputs to help batch and reduce reviews.

**Pull request closed without merging**